lib/cucloud/ec2_utils.rb

```diff
@@ -125,5 +142,68 @@ def instances_to_patch_by_tag(tag_name = 'auto_patch', tag_value = ['1'])
 
       all_instances
     end
+
+    # Get the nice name of the ec2 intsance from the 'Name" tag'
+    # @param instance_id [String] instance id in the format of i-121231231231
+    # @return [String] Name of instacnce if found or nil if not found
+    def get_instance_name(instance_id)
+      instance = get_instance(instance_id)
+      tag_name = instance.tags.find { |tag| tag.key.eql?('Name') }
+      tag_name ? tag_name.value : nil
+    end
+
+    # Create a snapshot of an EBS volume and apply supplied tags
+    # will wait 20 minutes for the process to completed
+    # @param volume_id [String] volume id in the formate of vol-121231231231
+    # @param snapshot_desc [String] Description of the snapshot
+    # @param tags [Array] Array of key value pairs to be applied as tags to the snapshot
+    # @return snapshot information see
+    # http://docs.aws.amazon.com/sdkforruby/api/Aws/EC2/Client.html#create_tags-instance_method
+    def create_ebs_snapshot(volume_id, snapshot_desc, tags = [])
+      snapshot_info = @ec2.create_snapshot(
+        volume_id: volume_id,
+        description: snapshot_desc
+      )
+
+      @ec2.wait_until(:snapshot_completed, snapshot_ids: [snapshot_info.snapshot_id]) do |w|
+        w.max_attempts = WAITER_MAX_ATTEMPS
+        w.delay = WAITER_DELAY
+      end
+
+      @ec2.create_tags(resources: [snapshot_info.snapshot_id], tags: tags) unless tags.empty?
+
+      snapshot_info
+    end
+
+    # Preforms a backup on volumes that do not have a recent snapshot_info
+    # @param days [Integer] defaults to 5
+    def backup_volumes_unless_recent_backup(days = 5)
+      volumes_backed_up_recently = volumes_with_snapshot_within_last_days(days)
+
+      volumes = @ec2.describe_volumes(filters: [{ name: 'attachment.status', values: ['attached'] }])
+      volumes.volumes.each do |volume|
+        next if volumes_backed_up_recently[volume.volume_id.to_s]
+        puts "backing up #{volume.volume_id}"
```

Comment:
Perhaps return an array of backed up instance ids so that the calling app can choose the reporting method?